static/js/components/EducationForm.js

```diff
@@ -180,27 +221,174 @@ class EducationForm extends ProfileFormFields {
     );
   }
 
-  render() {
-    let {
-      profile,
-      ui: { showEducationDeleteDialog }
+  clearEducationEdit: Function = (): void => {
+    const {
+      setEducationDialogVisibility,
+      setEducationDegreeLevel,
+      setEducationDialogIndex,
+      clearProfileEdit,
+      profile: { username },
     } = this.props;
+    setEducationDialogVisibility(false);
+    setEducationDegreeLevel('');
+    setEducationDialogIndex(null);
+    clearProfileEdit(username);
+  };
+
+  saveEducationForm: Function = (): void => {
+    const { saveProfile, profile, ui } = this.props;
+    saveProfile(educationValidation, profile, ui).then(() => {
+      this.clearEducationEdit();
+    });
+  };
 
-    if (profile['education'] === undefined) {
-      return null;
+  renderEducationEntries: Function = (): React$Element<*>[] => {
+    const { profile, profile: { education }} = this.props;
+    let rows = [];
+    if (education !== undefined) {
+      let sorted = educationEntriesByDate(education);
+      rows = sorted.map( ([index, entry]) => this.educationRow(entry, index));
     }
```

Comment:
what do you think about pulling this stuff out to a separate method? If we refactored `educationRow` to take an `[index, entry]` tuple we could declare a method:

```js
const sortedEducationRows = R.compose(R.map(this.educationRow), educationEntriesByDate);
```

or without ramda:


```js
const sortedEducationRows = entries => (
  educationEntriesByDate(entries).map(this.educationRow);
);
```